tutorials/Python/neonUtilities/neonUtilitiesPython.md

```diff
@@ -0,0 +1,342 @@
+
+
+```python
+base.setwd('/Downloads');
+```
+
+Run the downloader with these inputs: a DPID, a site (or "all" for all sites), a package (either basic or expanded), and an indicator to check the size of your download before proceeding, or not.
+
+There are two differences relative to running this function in R directly: 
+
+1. check.size becomes check_size, because dots have programmatic meaning in Python
+2. TRUE (or T) becomes "TRUE" because the values TRUE and FALSE don't have special meaning in Python the way they do in R, so it interprets them as variables if they're unquoted.
+
+check_size="TRUE" will estimate the size of the download and ask you to confirm before proceeding. This will cause problems in certain environments, or in batch processing. Under those circumstances, set check_size="FALSE", but consider the size of your query before doing this.
+
+
+```python
+neonUtils.zipsByProduct(dpID='DP1.10023.001', site='HARV', package='basic', check_size='TRUE');
+```
+
+    Continuing will download files totaling approximately 0.165245 MB. Do you want to proceed y/n: y
+    6 zip files downloaded to /Users/clunch/filesToStack10023
+    
+
+
+The message output by `zipsByProduct()` indicates the file path where the files have been downloaded.
+
+Now take that file path and pass it to `stackByTable()`. The file structure is slightly different from the zip file returned by the Portal, so we need an additional input, folder="TRUE".
+
+
+```python
+neonUtils.stackByTable("/Downloads/filesToStack10023", folder="TRUE");
+```
+
+    Unpacked  NEON.D01.HARV.DP1.10023.001.2013-07.basic.20180226T180545Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10023.001.2014-07.basic.20180226T174946Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10023.001.2015-06.basic.20180226T174941Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10023.001.2015-07.basic.20180226T175005Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10023.001.2016-07.basic.20180226T174902Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10023.001.2017-07.basic.20180226T174924Z.zip
+    
+    Finished: All of the data are stacked into  2  tables!
+    
+    Copied the first available variable definition file to /stackedFiles and renamed as variables.csv
+    Copied the first available validation file to /stackedFiles and renamed as validation.csv
+    Stacked  hbp_massdata
+    Stacked  hbp_perbout
+    
+
+
+## Download remote sensing files
+
+The function `byFileAOP()` uses the <a href="http://data.neonscience.org/data-api" target="_blank">NEON API</a> to programmatically download data files for remote sensing (AOP) data products. These files cannot be stacked by `stackByTable()` because they are not tabular data. The function simply creates a folder in your working directory and writes the files there. It preserves the folder structure for the subproducts.
```

Comment:
General question: Will this function work for the bundles eddy covariance products in HDF5 format too?

Reply:
Not right now, that's functionality we could potentially add in the future. We haven't addressed the EC products in neonUtilities yet.

Reply:
Okay.  Just wondering if it is designed for the non-csv products if it wouldn't be better to rename now, while there is other renaming going on and before the publicity push, to something without AOP in it.